### PR TITLE
support aten::index.Tensor

### LINKED ISSUE #766: ✨[Feature] Some operators need to be supported in Torch-TensorRT
**Is your feature request related to a problem? Please describe.**

  There are some operators need to be supported in Torch-TensorRT. It is from a Video-Swin-Transformer model.
  - aten::index.Tensor(Tensor self, Tensor?[] indices) -> (Tensor)
  - aten::adaptive_avg_pool3d(Tensor self, int[3] output_size) -> (Tensor)
  - aten::Int.Tensor(Tensor a) -> (int)
  - aten::remainder.Tensor(Tensor self, Tensor other) -> (Tensor)
  - aten::div.Tensor_mode(Tensor self, Tensor other, *, str? rounding_mode) -> (Tensor)
  - aten::remainder.Scalar(Tensor self, Scalar other) -> (Tensor)
  - aten::roll(Tensor self, int[1] shifts, int[1] dims=[]) -> (Tensor)
  - aten::to.dtype_layout(Tensor(a) self, *, int? dtype=None, int? layout=None, Device? device=None, bool? pin_memory=None, bool non_blocking=False, bool copy=False, int? memory_format=None) -> (Tensor(a))

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by ruoqianguo**
# Description

`aten::index.Tensor(Tensor self, Tensor?[] indices) -> (Tensor)` is a complicated PyTorch function which can receive several indices. To focus on Video Swin Transformer, this version converter only supports one index. I will improve this converter in a future release.  

Fixes #766 

## Type of change

Please delete options that are not relevant and/or add your own.

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [x] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: core/conversion/converters/impl/select.cpp [hunk 1]
```diff
@@ -202,6 +230,65 @@ auto select_registrations TORCHTRT_UNUSED =
...
+                        tensors.push_back(tensor_to_const(ctx, torch_tensor));
+                      } else {
+                        auto cont = t.toCustomClass<TensorContainer>();
+                        tensors.push_back(cont->tensor());
+                      }
+                    }
+
+                    TORCHTRT_CHECK(
+                        tensors.size() == 1,
+                        "This version of Torch-TensorRT only supports one index in aten::index.Tensor");
```
Comment (narendasan) [inline core/conversion/converters/impl/select.cpp:274]: Can you clarify what this means? Can I only do something like tensor[0] or something?

Comment (ruoqianguo) [reply core/conversion/converters/impl/select.cpp:274]: In Pytorch, 'aten::index.Tensor' can receive several indices, like this:
```
x = torch.rand(5, 7, 10)
idx0 = torch.tensor([0, 0, 0, 0])
idx1 = torch.tensor([1, 1, 1, 1])
idx2 = torch.tensor([2, 2, 2, 2])
y1 = x[idx0, idx1, idx2] # result is like [x[0, 1, 2], x[0, 1, 2], x[0, 1, 2], x[0, 1, 2]]
y2 = x[idx0] # result is like [x[0, :, :], x[0, :, :], x[0, :, :], x[0, :, :]]
```
I only support one index as input currently like `y2 = x[idx0]`

Comment (ruoqianguo) [reply core/conversion/converters/impl/select.cpp:274]: > In Pytorch, 'aten::index.Tensor' can receive several indices, like this:
> 
> ```
> x = torch.rand(5, 7, 10)
> idx0 = torch.tensor([0, 0, 0, 0])
> idx1 = torch.tensor([1, 1, 1, 1])
> idx2 = torch.tensor([2, 2, 2, 2])
> y1 = x[idx0, idx1, idx2] # result is like tensor([x[0, 1, 2], x[0, 1, 2], x[0, 1, 2], x[0, 1, 2]])
> y2 = x[idx0] # result is like tensor([x[0, :, :], x[0, :, :], x[0, :, :], x[0, :, :]])
> ```
> 
> I only support one index as input currently like `y2 = x[idx0]`. We can only process tensors[0].

Hi @narendasan Is this explanation clear?

Comment (narendasan) [reply core/conversion/converters/impl/select.cpp:274]: Ok can we improve this message to convey what this means

Comment (ruoqianguo) [reply core/conversion/converters/impl/select.cpp]: Hi @narendasan i have added some annotation and improved this message.

**Review by narendasan — APPROVED** [review]
LGTM

**Comment by ncomly-nvidia** [comment]
Fixes #468

**Pull request merged**